examples/Remarketing/UploadStoreSalesTransactions.cs

```diff
@@ -0,0 +1,488 @@
+                externalId: externalId,
+                advertiserUploadDateTime: advertiserUploadDateTime,
+                bridgeMapVersionId: bridgeMapVersionId,
+                partnerId: partnerId);
+        }
+
+        /// <summary>
+        /// Returns a description about the code example.
+        /// </summary>
+        public override string Description
+        {
+            get
+            {
+                return "This code example uploads offline data for store sales transactions. " +
+                       "This feature is only available to allowlisted accounts. " +
+                       "See https://support.google.com/google-ads/answer/7620302 for more details.";
+            }
+        }
+
+        /// <summary>
+        /// Runs the code example.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <param name="conversionActionId">The ID of a store sales conversion action.</param>
+        /// <param name="offlineUserDataJobType">The type of user data in the job (first or third
+        ///     party). If you have an official store sales partnership with Google, use
+        ///     StoreSalesUploadThirdParty. Otherwise, use StoreSalesUploadFirstParty or
+        ///     omit this parameter.</param>
+        /// <param name="externalId">Optional (but recommended) external ID to identify the offline
+        ///     user data job.</param>
+        /// <param name="advertiserUploadDateTime">Date and time the advertiser uploaded data to the
+        ///     partner. Only required if uploading third party data.</param>
+        /// <param name="bridgeMapVersionId">Version of partner IDs to be used for uploads. Only
+        ///     required if uploading third party data.</param>
+        /// <param name="partnerId">ID of the third party partner. Only required if uploading third
+        ///     party data.</param>
+        public void Run(GoogleAdsClient client, long customerId, long conversionActionId,
+            OfflineUserDataJobType offlineUserDataJobType =
+                OfflineUserDataJobType.StoreSalesUploadFirstParty,
+            long? externalId = null, string advertiserUploadDateTime = null,
+            string bridgeMapVersionId = null, long? partnerId = null)
+        {
+            // Get the CampaignService.
+            OfflineUserDataJobServiceClient offlineUserDataJobServiceClient =
+                client.GetService(Services.V3.OfflineUserDataJobService);
+
+            // Ensure that a valid job type is provided.
+            if (offlineUserDataJobType != OfflineUserDataJobType.StoreSalesUploadFirstParty &
+                offlineUserDataJobType != OfflineUserDataJobType.StoreSalesUploadThirdParty)
+            {
+                Console.WriteLine("Invalid job type specified, defaulting to First Party.");
```

Comment:
Not sure if there's precedence for this elsewhere in the .NET examples, but I'd prefer we throw an exception in this case.

Reply:
Anash and I discussed this briefly. This is a minor hack that has to do with the way the .NET example solution is set up; the example runner code will supply the first value from the enum, which happens to be .Unspecified. This overrides the default value of .FirstParty and breaks the example.

It wouldn't be an issue if running this file directly via command line.

Reply:
OK, makes sense. Thanks for the context.